[CRM-262] feat: QR코드 카메라 인식 기능 및 사용 처리 로직 수정

bonyubking opened the pull request:
[feat: QR코드 카메라 인식 기능 및 사용 처리 로직 수정

{token}/use api를 통해 카메라로 QR 촬영시 사용 로직 수정
현재는 adminId만 가능함(admin_code_Id 검증없음)

{token}/verify 를 통해 검증 후 {toekn}/use 호출

bonyubking committed:
feat: 마이페이지 - 예매 내역 보기 실시간 혼잡도 조회 버튼 추가

내 예매내역에 티켓 이름 추가

bonyubking committed:
Merge branch 'fix/mypage-ads' into CRM-262-qrcode-camera

bonyubking committed:
feat: QR코드 카메라 인식 기능 및 사용 처리 로직 수정

{token}/use api를 통해 카메라로 QR 촬영시 사용 로직 수정
현재는 adminId만 가능함(admin_code_Id 검증없음)

{token}/verify 를 통해 검증 후 {toekn}/use 호출

choi2h commented on src/main/java/com/myce/qrcode/service/mapper/QrResponseMapper.java:
```diff
@@ -0,0 +1,56 @@
+        
+        switch (status) {
+            case APPROVED:
+                return QrVerifyResponse.invalid("QR 코드가 활성화되지 않았습니다.", "APPROVED");
```
이거 status에 message필드 추가해서 사용해주세요

choi2h commented on src/main/java/com/myce/qrcode/service/mapper/QrResponseMapper.java:
```diff
@@ -0,0 +1,56 @@
+                        reserver.getReservation().getTicket().getName()
+                );
+                
+            default:
```
이건 에러로 반환되어야하지 않ㅇ을까요...?

choi2h commented on src/main/java/com/myce/qrcode/service/mapper/QrResponseMapper.java:
```diff
@@ -0,0 +1,56 @@
+            case ACTIVE:
+                // 유효한 QR 코드 - 예약자 정보와 함께 반환
+                Reserver reserver = qrCode.getReserver();
+                return QrVerifyResponse.valid(
```
이것도 ACTIVE에 대한 메시지로 분리해서 주입해주세요

choi2h commented on src/main/java/com/myce/qrcode/service/impl/QrCodeServiceImpl.java:
```diff
@@ -88,7 +92,7 @@ public void reissueQr(Long reserverId, Long adminMemberId) {
     @Override
     @Transactional
-    public void markQrAsUsed(String qrToken, Long adminMemberId) {
+    public QrUseResponse markQrAsUsed(String qrToken, Long adminMemberId) {
```
mark말고 수정에 대한건 update로 통일하기로 했습니다! 수정해주세요!

bonyubking committed:
feat: mark -> update 수정 및 message 클래스 처리

choi2h commented on src/main/java/com/myce/qrcode/controller/QrCodeController.java:
```diff
@@ -31,10 +35,11 @@ public ResponseEntity<Void> reissue(@PathVariable Long reserverId,
     }
 
     @PostMapping("/token/{token}/use")
```
여기 Controller HttpMethod 다시 확인해주세요~

bonyubking committed:
fix: QR 코드 사용과 검증 @PathVariable -> @RequestBody (보안)

 향후 확장성을 위해 Token 값 자체가 아닌 QrTokenRequest 에 저장

bonyubking committed:
fix: QR 코드 사용과 검증 @PathVariable -> @RequestBody (보안)

 향후 확장성을 위해 Token 값 자체가 아닌 QrTokenRequest 에 저장

bonyubking committed:
fix: QR 코드 사용과 검증 @PathVariable -> @RequestBody (보안)

 향후 확장성을 위해 Token 값 자체가 아닌 QrTokenRequest 에 저장

bonyubking committed:
fix: QR 코드 사용과 검증 @PathVariable -> @RequestBody (보안)

 향후 확장성을 위해 Token 값 자체가 아닌 QrTokenRequest 에 저장

bonyubking committed:
fix: QR 코드 사용과 검증 @PathVariable -> @RequestBody (보안)

 향후 확장성을 위해 Token 값 자체가 아닌 QrTokenRequest 에 저장

 fix : mypage/reservationdetail

 pathvariable memberId로 수정

bonyubking committed:
Merge remote-tracking branch 'origin/develop' into CRM-262-qrcode-camera

bonyubking committed:
fix: QR 코드 사용과 검증 @PathVariable -> @RequestBody (보안)

 향후 확장성을 위해 Token 값 자체가 아닌 QrTokenRequest 에 저장

 fix : mypage/reservationdetail

 pathvariable memberId로 수정
